airflow/executors/python_executor.py

```diff
@@ -0,0 +1,173 @@
+
+    Does'nt start new process!
+    """
+    _is_terminated = threading.Event()
+
+    def __init__(self, fail_fast=False):
+        super(PythonExecutor, self).__init__()
+        self.tasks_to_run = []
+        self.fail_fast = fail_fast
+
+    def execute_async(self, key, command, queue=None, executor_config=None):
+        self.tasks_to_run.append((key, command))
+
+    def sync(self):
+        task_failed = False
+        for key, command in self.tasks_to_run:
+            ti = command[0]
+            if self.fail_fast and task_failed:
+                logger.info("Setting %s to %s", key, State.UPSTREAM_FAILED)
+                ti.set_state(State.UPSTREAM_FAILED)
+                self.change_state(key, State.UPSTREAM_FAILED)
+                continue
+
+            if self._is_terminated:
+                logger.info(
+                    "Executor is terminated! Stopping %s to %s", key, State.FAILED
+                )
+                ti.set_state(State.FAILED)
+                self.change_state(key, State.FAILED)
+                continue
+
+            self.log.debug("Executing task: %s", command[0])
+
+            try:
+                self._run_task_instance(ti, **command[1])
+
+                self.change_state(key, State.SUCCESS)
+            except subprocess.CalledProcessError as e:
+                task_failed = True
+                self.change_state(key, State.FAILED)
+                self.log.error("Failed to execute task: %s.", str(e))
+            except AirflowException as e:
+                task_failed = True
+                self.change_state(key, State.FAILED)
+                self.log.error("Failed to execute task: %s.", str(e))
+            except KeyboardInterrupt as e:
+                task_failed = True
+                fail_fast = True
+                self.change_state(key, State.FAILED)
+                self.log.exception("Interrupted to execute task: %s.", str(e))
+            except Exception as e:
+                task_failed = True
+                self.change_state(key, State.FAILED)
+                self.log.exception("Failed to execute task: %s.", str(e))
+
+        self.tasks_to_run = []
+
+    def queue_task_instance(
+        self,
+        task_instance,
+        mark_success=False,
+        pickle_id=None,
+        ignore_all_deps=False,
+        ignore_depends_on_past=False,
+        ignore_task_deps=False,
+        ignore_ti_state=False,
+        pool=None,
+        cfg_path=None,
+    ):
+
+        self.queue_command(
+            task_instance,
+            (
+                task_instance,
+                dict(
+                    mark_success=mark_success,
+                    # ignore_all_deps =ignore_all_deps,
+                    # ignore_depends_on_past=ignore_depends_on_past,
+                    # ignore_task_deps=ignore_task_deps,
+                    # ignore_ti_state=ignore_ti_state,
+                    pool=pool,
+                    # pickle_id=pickle_id,
```

Comment:
Should we remove this comment together with unused params?

Reply:
Check original Sequential, probably we need to uncomment